test/analytics/__fixtures__/plugins/analytics_plugin_a/public/plugin.ts

```diff
@@ -52,10 +52,20 @@ export class AnalyticsPluginA implements Plugin {
       plugin: 'analyticsPluginA',
       step: 'setup',
     });
-
+    let found = false;
     window.__analyticsPluginA__ = {
-      getLastActions: async (takeNumberOfActions: number) =>
-        firstValueFrom(this.actions$.pipe(take(takeNumberOfActions), toArray())),
+      getLastActions: async () =>
```

Comment:
nit: we might want to rename this method to `getActionsUntilReportTestPluginLifecycleEvent`. What do you think?